web/src/app/util/RequireConfig.tsx

```diff
@@ -102,44 +116,48 @@ export function useConfig() {
 // ```js
 // const [mailgun, slack] = useConfigValue('Mailgun.Enable', 'Slack.Enable')
 // ```
-export function useConfigValue(...fields) {
+export function useConfigValue(...fields: ConfigID[]): Value[] {
   const config = useConfig()
   return fields.map((f) => config[f])
 }
 
-export function Config({ children }) {
-  const { isAdmin, userID, config } = useContext(ConfigContext)
-  return children(mapConfig(config), { isAdmin, userID }) || null
+export function Config(props: {
+  children: (x: ConfigData, s?: SessionInfo) => JSX.Element
+}): JSX.Element {
+  return props.children(useConfig(), useSessionInfo()) || null
 }
 
-export default function RequireConfig(props) {
+export type RequireConfigProps = {
+  configID: ConfigID
+  // test to determine whether or not children or else is returned
+  test?: (x: Value) => boolean
+
+  // react element to render if checks failed
+  else?: JSX.Element
+  isAdmin?: boolean
+  children: React.ReactChildren
+}
+
+export default function RequireConfig(
+  props: RequireConfigProps,
+): JSX.Element | null {
   const {
     configID,
     test = isTrue,
     isAdmin: wantIsAdmin,
     children,
     else: elseValue = null,
   } = props
-  const { config, isAdmin } = useContext(ConfigContext)
+  const config = useConfig()
+  const { isAdmin } = useSessionInfo()
 
   if (wantIsAdmin && !isAdmin) {
     return elseValue
   }
 
-  const cfgOpt = config.find((c) => c.id === configID)
-  if (configID && !test(cfgOpt?.value)) {
+  if (configID && !test(config[configID])) {
```

Comment:
We want the changes from the previous PR:

```suggestion
  const cfgOpt = config.find((c) => c.id === configID)
  if (configID && !test(cfgOpt?.value)) {
```